pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -394,7 +394,9 @@ func linkGoPackage(root string, pkg *schema.Package, out string) error {
 		return err
 	}
 	if pkg.Parameterization == nil {
-		relOut = filepath.Join(relOut, pkg.Name)
+		// Go SDK Gen replaces all "-" in the name.  See pkg/codegen/gen.go:goPackage
+		name := strings.ReplaceAll(pkg.Name, "-", "")
```

Comment:
Guessing there's no way we can reuse that functionality instead of making sure this stays in-sync with it?